wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -495,6 +502,55 @@ public void createSnapshot(String workspaceId) throws NotFoundException, ServerE
         return machineManager.getSnapshots(sessionUser().getUserId(), workspaceId);
     }
 
+    /**
+     * Creates and adds new machine into specified workspace runtime.
+     * If machine creation fails, then that machine will be destroyed.
+     *
+     * @param machineConfig
+     *         configuration of new machine
+     * @param workspaceId
+     *         id of workspace this machine belongs to
+     * @param activeEnv
+     *         current machine environment
+     * @return starting machine instance
+     * Throws the same exceptions as @{@link MachineManager#createMachineAsync(MachineConfig, String, String)}
+     */
+    public MachineImpl createMachineAsyncInRuntime(MachineConfig machineConfig, String workspaceId, String activeEnv)
+            throws MachineException,
+                   BadRequestException,
+                   SnapshotException,
+                   NotFoundException,
+                   ConflictException {
+        new AddNewMachineInRuntimeEventSubscriber();
```

Comment:
This line looks really strange for Java. Name of class looks like name of method. Please refactor